client/my-sites/plugins/marketplace/marketplace-plugin-details/test/index.js

```diff
@@ -55,22 +60,52 @@ describe( '<PurchaseArea/> Plugin details next step tests', () => {
 				onAddYoastPremiumToCart={ onAddYoastPremiumToCart }
 				onNavigateToCheckout={ onNavigateToCheckoutMockFunction }
 				onNavigateToDomainsSelection={ onNavigateToDomainsSelectionMockFunction }
+				onRemoveEverythingFromCart={ onRemoveEverythingFromCart }
 			/>
 		);
 		const yoastPremiumButton = await marketplacePluginDetails.findByText( 'Buy Yoast Premium' );
-		const yoastFreeButton = await marketplacePluginDetails.findByText( 'Add Yoast Free' );
-
 		await fireEvent.click( yoastPremiumButton );
+		//onAddPlugin call in the PurchaseArea component is async so we have to queue and await a promise  for the following mocks to be called
+		await Promise.resolve();
+
 		expect( onAddYoastPremiumToCart ).toHaveBeenCalled();
 		expect( onNavigateToCheckoutMockFunction ).not.toHaveBeenCalled();
 		expect( onNavigateToDomainsSelectionMockFunction ).toHaveBeenCalled();
 		await onAddYoastPremiumToCart.mockReset();
 		await onNavigateToCheckoutMockFunction.mockReset();
 		await onNavigateToDomainsSelectionMockFunction.mockReset();
 
+		const yoastFreeButton = await marketplacePluginDetails.findByText( 'Add Yoast Free' );
 		await fireEvent.click( yoastFreeButton );
+		//onAddPlugin call in the PurchaseArea component is async so we have to queue and await a promise  for the following mocks to be called
+		await Promise.resolve();
+
 		expect( onAddYoastPremiumToCart ).not.toHaveBeenCalled();
 		expect( onNavigateToCheckoutMockFunction ).not.toHaveBeenCalled();
 		expect( onNavigateToDomainsSelectionMockFunction ).toHaveBeenCalled();
 	} );
+
+	test( 'Remove all other products from basket when adding a marketplace plugin', async () => {
+		const marketplacePluginDetails = render(
+			<PurchaseArea
+				siteDomains={ [ primaryWpcomDomain ] }
+				isProductListLoading={ false }
+				displayCost={ '$ 100' }
+				wporgPluginName={ 'Yoast SEO' }
+				onAddYoastPremiumToCart={ onAddYoastPremiumToCart }
+				onNavigateToCheckout={ onNavigateToCheckoutMockFunction }
+				onNavigateToDomainsSelection={ onNavigateToDomainsSelectionMockFunction }
+				onRemoveEverythingFromCart={ onRemoveEverythingFromCart }
+			/>
+		);
+		const yoastPremiumButton = await marketplacePluginDetails.findByText( 'Buy Yoast Premium' );
+		const yoastFreeButton = await marketplacePluginDetails.findByText( 'Add Yoast Free' );
+
+		await fireEvent.click( yoastPremiumButton );
+		expect( onRemoveEverythingFromCart ).toHaveBeenCalled();
+		await onRemoveEverythingFromCart.mockReset();
+
+		await fireEvent.click( yoastFreeButton );
+		expect( onRemoveEverythingFromCart ).toHaveBeenCalled();
+	} );
```

Comment:
Great job on maintaining the test suite! However, there seems to be something wrong. 
Running `yarn test-client client/my-sites/plugins/marketplace/marketplace-plugin-details` gave me 3 tests passed. However, when I commented out the line 104, 106 and 108, it still gave me 3 tests passed, which meant the passing conditions weren't working. I don't know why, though.